test/src/de/dkfz/b080/co/knowledge/metadata/COMetadataAccessorSpec.groovy

```diff
@@ -42,4 +61,38 @@ class COMetadataAccessorSpec extends Specification {
         }
     }
 
+    def "Extract sample name from BAM basename"(String filename, String possibleControlSampleNamePrefixes, String possibleTumorSampleNamePrefixes, boolean searchWithSeparator, String resultSample) {
+
+        when:
+        context.configurationValues.add(new ConfigurationValue(COConstants.CVALUE_POSSIBLE_CONTROL_SAMPLE_NAME_PREFIXES, possibleControlSampleNamePrefixes, ConfigurationConstants.CVALUE_TYPE_BASH_ARRAY))
+        context.configurationValues.add(new ConfigurationValue(COConstants.CVALUE_POSSIBLE_TUMOR_SAMPLE_NAME_PREFIXES, possibleTumorSampleNamePrefixes, ConfigurationConstants.CVALUE_TYPE_BASH_ARRAY))
+        context.configurationValues.add(new ConfigurationValue(COConstants.CVALUE_SEARCH_MERGEDBAM_WITH_SEPARATOR, searchWithSeparator.toString(), ConfigurationConstants.CVALUE_TYPE_BOOLEAN))
+
+        then:
+        (new COMetadataAccessor(new BasicCOProjectsRuntimeService()).extractSampleNameFromBamBasename(filename, context)) == resultSample
+
+        where:
+        filename                          | possibleControlSampleNamePrefixes | possibleTumorSampleNamePrefixes | searchWithSeparator | resultSample
+        "control_TEST002_mdup.bam"        | "( control )"                     | "( )"                           | false               | "control"
+        "control_abc_TEST003_mdup.bam"    | "( control_abc )"                 | "( )"                           | false               | "control_abc"
+        "control_abc_TEST004_mdup.bam"    | "( control control_abc )"         | "( )"                           | false               | "control_abc"
+        "control_abc_0_TEST005_mdup.bam"  | "( control control_abc_0 )"       | "( )"                           | false               | "control_abc_0"
+        "control_abc_01_TEST006_mdup.bam" | "( control control_abc_0 )"       | "( )"                           | false               | "control_abc_01"
+        "control_abc_01_TEST007_mdup.bam" | "( control control_abc_0 )"       | "( )"                           | true                | "control"
+        "control_02_TEST008_mdup.bam"     | "( control_02 )"                  | "( )"                           | false               | "control_02"
+        "Control_02_TEST009_mdup.bam"     | "( control_02 )"                  | "( )"                           | false               | "Control_02"
+        "CONTROL_02_TEST010_mdup.bam"     | "( control_02 )"                  | "( )"                           | false               | "CONTROL_02"
+
+        // In the following cases, the sample extraction will "Fail". But this is not a fault of the code, we just do not have enough information to
+        // extract more. Maybe later, this could be extended with regex etc. But for now it is not possible.
+        "control_abc_02_TEST011_mdup.bam" | "( control )"                     | "( )"                           | false               | "control"
```

Comment:
Would you add a case "( control )" with a file "control_02_TEST13_mdup.bam" (remember -- we had a test for integer before)? Actually, I think that the "failed" matches (maybe better "unintentional" matches) are very insightful because they show he limitations of the algorithm.

Reply:
With which result?
control_02?

Reply:
Yes, we had the test before for 1/2-digits after the sample name for non atomic sample names. We should check it.